### ISSUE TITLE
Remote partition processor auto-detection fails when ReFrame is installed from PyPI

### ISSUE THREAD

**Issue by jack-morrison** [reframe-hpc/reframe#2914]
To reproduce, install ReFrame via `pip install reframe-hpc` and launch a job with remote detection enabled on a partition unknown to ReFrame in `~/.reframe/topology`.

``` [2023-06-08T09:27:21] warning: reframe: ^[[33mWARNING: failed to retrieve remote processor info: [Errno 2] No such file or directory: '/home/user/venv/lib/python3.10/site-packages/bin/'^[[0m
[2023-06-08T09:27:21] debug: reframe: Traceback (most recent call last):
  File "/home/user/venv/lib/python3.10/site-packages/reframe/frontend/autodetect.py", line 142, in _remote_detect
    with _copy_reframe(prefix) as dirname:
  File "/home/user/venv/lib/python3.10/site-packages/reframe/frontend/autodetect.py", line 57, in __enter__
    shutil.copy2(src, self._workdir)
  File "/home/user/Python-3.10.11/build/lib/python3.10/shutil.py", line 434, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/home/user/Python-3.10.11/build/lib/python3.10/shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/venv/lib/python3.10/site-packages/bin/'
```

It appears that the remote partition processor auto-detection expects a certain installation structure, and the structure produced when installing ReFrame via `pip` is not compatible with that expectation. As a result, ReFrame is unable to produce remote partition topology files.

**Comment by vkarak** [comment]
I could reproduce it by installing `reframe-hpc` in a venv.

**Comment by satishskamath** [comment]
I installed ReFrame via Easybuild and am getting this:
```
Detecting topology of remote partition 'snellius:gpu': this may take some time...
WARNING: failed to retrieve remote processor info: [Errno 2] No such file or directory: '/home/satishk/.local/easybuild/RHEL8/2022/software/ReFrame/4.2.0/lib/python3.6/site-packages/bin/'
Traceback (most recent call last):
  File "/home/satishk/.local/easybuild/RHEL8/2022/software/ReFrame/4.2.0/lib/python3.6/site-packages/reframe/frontend/autodetect.py", line 142, in _remote_detect
    with _copy_reframe(prefix) as dirname:
  File "/home/satishk/.local/easybuild/RHEL8/2022/software/ReFrame/4.2.0/lib/python3.6/site-packages/reframe/frontend/autodetect.py", line 57, in __enter__
    shutil.copy2(src, self._workdir)
  File "/usr/lib64/python3.6/shutil.py", line 263, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib64/python3.6/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/home/satishk/.local/easybuild/RHEL8/2022/software/ReFrame/4.2.0/lib/python3.6/site-packages/bin/'
```

**Comment by vkarak** [comment]
@satishskamath This will be fixed by #2978.
